File: C programs/104. Maximum Depth of Binary Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int maxDepth(struct TreeNode* root){
    if(root==NULL)
    {
        return NULL;
    }

    int left_d=maxDepth(root->left);
    int right_d=maxDepth(root->right);


   return 1 + (right_d>left_d ? right_d:left_d);
}

File: other languages/104. Maximum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        def dfs(root,d):
            if not root: return d
            return max(dfs(root.left,d+1),dfs(root.right,d+1))
        return dfs(root,0)
